core/src/main/java/tc/oc/pgm/channels/MessageChannel.java

```diff
@@ -0,0 +1,292 @@
+        || value.equals(SettingValue.SOUNDS_CHAT)
+        || value.equals(SettingValue.SOUNDS_DM)) target.playSound(SOUND);
+
+    Bukkit.getPluginManager()
+        .callEvent(
+            new ChannelMessageEvent<MatchPlayer>(
+                event.getChannel(),
+                sender,
+                event.getTarget(),
+                Collections.singletonList(sender),
+                event.getMessage()));
+  }
+
+  @Override
+  public Component formatMessage(
+      MatchPlayer target, @Nullable MatchPlayer sender, Component message) {
+    if (!target.equals(sender))
+      return text()
+          .append(translatable("misc.to", NamedTextColor.GRAY, TextDecoration.ITALIC))
+          .append(space())
+          .append(target.getName(NameStyle.VERBOSE))
+          .append(text(": ", NamedTextColor.WHITE))
+          .append(message)
+          .build();
+
+    return text()
+        .append(translatable("misc.from", NamedTextColor.GRAY, TextDecoration.ITALIC))
+        .append(space())
+        .append(sender.getName(NameStyle.VERBOSE))
+        .append(text(": ", NamedTextColor.WHITE))
+        .append(message)
+        .build();
+  }
+
+  @Override
+  public void registerCommand(PaperCommandManager<CommandSender> manager) {
+    manager.command(
+        manager
+            .commandBuilder("msg", "tell")
+            .argument(
+                StringArgument.<CommandSender, MatchPlayer>ofType(
+                        TypeToken.get(MatchPlayer.class), "target")
+                    .withParser(
+                        manager
+                            .getParserRegistry()
+                            .createParser(
+                                TypeToken.get(MatchPlayer.class), ParserParameters.empty())
+                            .orElseThrow(IllegalStateException::new))
+                    .build())
+            .handler(
+                context -> {
+                  MatchPlayer sender =
+                      context.inject(MatchPlayer.class).orElseThrow(IllegalStateException::new);
+                  final MatchPlayer target = context.<MatchPlayer>get("target");
+                  setSelectedTarget(sender, target);
+                  PGM.get().getChannelManager().setChannel(sender, this);
+                }));
+
+    manager.command(
+        manager
+            .commandBuilder("msg", "tell")
+            .argument(
+                StringArgument.<CommandSender, MatchPlayer>ofType(
+                        TypeToken.get(MatchPlayer.class), "target")
+                    .withParser(
+                        manager
+                            .getParserRegistry()
+                            .createParser(
+                                TypeToken.get(MatchPlayer.class), ParserParameters.empty())
+                            .orElseThrow(IllegalStateException::new))
+                    .build())
+            .argument(
+                StringArgument.<CommandSender>builder("message")
```

Comment:
you can chain commands like these.

after the `.handler(context -> ...)` you can append the next .argument and another .handler. The first handler is used if the command is up to the first argument, the second is used when the command also has the 2nd param